inputremapper/gui/controller.py

```diff
@@ -244,8 +244,36 @@ def copy_preset(self):
         )
         self.message_broker.publish(DoStackSwitch(1))
 
+    def _auto_use_as_analog(self, combination: InputCombination) -> InputCombination:
+        """If output is analog, set the first fitting input to analog."""
+        if self.data_manager.active_mapping is None:
+            return combination
+
+        if not self.data_manager.active_mapping.is_analog_output():
+            return combination
+
+        if combination.find_analog_input_config():
+            # something is already set to do that
+            return combination
+
+        for i, input_config in enumerate(combination):
+            # find the first analog input and set it to "use as analog"
+            if input_config.type in (EV_ABS, EV_REL):
+                logger.info("Using %s as analog input", input_config)
+
+                # combinations and input_configs are immutable, a new combination
+                # is created to fit the needs instead
+                combination_list = list(combination)
+                combination_list[i] = input_config.modify(analog_threshold=0)
+                new_combination = InputCombination(combination_list)
+                return new_combination
+
+        return combination
+
     def update_combination(self, combination: InputCombination):
         """Update the input_combination of the active mapping."""
+        combination = self._auto_use_as_analog(combination)
+
```

Comment:
This is kinda confusing. If I call update_combination with some new combination I would expect the combination to be the one I provided and not a modified version of it.

I think it is better if we call `_auto_use_as_analog` directly in the methods where we are sure we want to apply the logic. Currently this seems to be only in the `_on_combination_recorded` method.

Reply:
Thanks